systemtest/tls_test.go

```diff
@@ -79,6 +75,51 @@ func TestTLSConfig(t *testing.T) {
 	})
 }
 
+func TestCustomTLSConfig(t *testing.T) {
```

Comment:
Would it make sense to also add tests that ensure non-fips configuration leads to failures? This is running the same test cases for fips and non-fips, but what about the cases that are allowed in non-fips but are expected to fail in fips?